### PR TITLE
feat: add pronouns to Person and oidc claims. Fixes #4043.

### LINKED ISSUE #4043: New pronouns field
### Description

As per the [consultation](https://mailarchive.ietf.org/arch/msg/ietf-announce/aweKuk52RkhA8DPPLMhC7BedRNU/), a new field should be added "Pronouns" that is shown on the public profile.  

The Pronouns field either allows the user to select any number of the following:
- he/him
- she/her
- they/them
Or they can select "Other" which is a free text field of up to 20 characters.

The informational text next to the field should be "Optionally provide your personal pronouns.  These will be displayed on your public profile page and alongside your name in Meetecho and, in future, other systems."

The Pronouns field should be available as a new OIDC claim that returns either those selected from the list with commas in between, or the contents of the free text field, depending on what was selected.

### Code of Conduct

- [X] I agree to follow the [IETF's Code of Conduct](https://github.com/ietf-tools/.github/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by rjsparks**

### CODE REVIEW

**Review by jennifer-richards — CHANGES_REQUESTED** [review]
This looks like it should work (and seems to, though I have not exercised the OIDC claim directly). A couple comments, though.

This implementation differs from the description in #4043. The specification there seems to ask for checkboxes with the standard options, allowing zero or more to be selected, plus one for the free-form entry. That strikes me as an odd format, but it's described specifically enough that I wonder if there's a particular demand for it?

UI-wise, I have mixed feelings about the hinted text field. The only hint to the user of the format expected comes from the autocomplete options that show up when they start typing. An old fashioned radio or checkbox would make it clearer, which might be especially helpful for people unfamiliar with the practice of specifying one's pronouns.

Also, the display of the pronouns on the profile page seems to emphasize the pronouns disproportionately. I.e., they're on their own line with a font size larger than all but the person's name.

Marking this as "Request changes," but it's really more of a request to consider/confirm the design change compared to the ticket (and probably a little bike shedding).

**Comment by rjsparks** [comment]
02cbf1a tries a different approach, using a MultiWidget (CheckboxSelectMultiple and TextInput), but the form mechanics are convoluted and it will still need more validation. Will try separate model (not just form) fields next.

File: ietf/ietfauth/forms.py [hunk 1]
```diff
@@ -152,12 +171,18 @@ def clean_ascii_short(self):
             prevent_system_name(name)
             return ascii_cleaner(name)
 
+        def clean_pronouns(self):
+            pronouns = self.cleaned_data.get("pronouns") or ""
+            pronouns = pronouns.strip()
```
[the remark this answers is not in the document]
Comment (rjsparks) [reply ietf/ietfauth/forms.py]: All the above is OBE - this code took a different path

**Review by jennifer-richards — APPROVED** [review]
The changes look good - the modeling is a little awkward, but seems like a good compromise given the constraints of Django's fields/widgets.

My only suggestion is that the display on the profile page still feels a little too "in your face". Playing with it, getting rid of the \<br> and dropping the font to fs-4 or -5 instead of fs-3 gives me something that seems a little more like a typical name badge level of attention to the pronouns.

Could iterate forever on aesthetics and page design, though, so I'm approving.

**Pull request merged**